core/src/main/java/nu/yona/server/device/entities/DeviceAnonymized.java

```diff
@@ -60,17 +62,20 @@ protected DeviceAnonymized()
 		super(null);
 	}
 
-	private DeviceAnonymized(UUID id, int deviceIndex, OperatingSystem operatingSystem, String appVersion)
+	private DeviceAnonymized(UUID id, int deviceIndex, OperatingSystem operatingSystem, String appVersion,
+			Optional<String> firebaseInstanceId)
 	{
 		super(id);
 		this.deviceIndex = deviceIndex;
 		this.operatingSystem = operatingSystem;
 		this.appVersion = Objects.requireNonNull(appVersion);
+		this.firebaseInstanceId = Objects.requireNonNull(firebaseInstanceId).orElse(null);
```

Comment:
The call to ``Objects.requireNonNull`` doesn't add value here. If ``firebaseInstanceId`` is null, ``Objects.requireNonNull`` would throw an NPE, but that would also occur because of the call to ``orElse``.

Reply:
Simplified